components/public-api-server/pkg/proxy/conn.go

```diff
@@ -42,3 +46,69 @@ func (p *NoConnectionPool) Get(ctx context.Context, token string) (gitpod.APIInt
 
 	return server, nil
 }
+
+func NewConnectionPool(address *url.URL, poolSize int) (*ConnectionPool, error) {
+	cache, err := lru.NewWithEvict(poolSize, func(key, value interface{}) {
+		connectionPoolSize.Dec()
+
+		// We attempt to gracefully close the connection
+		conn, ok := value.(gitpod.APIInterface)
+		if !ok {
+			return
+		}
+
+		closeErr := conn.Close()
+		if closeErr != nil {
+			log.Log.WithError(closeErr).Warn("Failed to close connection to server.")
+		}
+	})
+	if err != nil {
+		return nil, fmt.Errorf("faield to create LRU cache: %w", err)
+	}
+
+	return &ConnectionPool{
+		cache: cache,
+		connConstructor: func(token string) (gitpod.APIInterface, error) {
+			return gitpod.ConnectToServer(address.String(), gitpod.ConnectToServerOpts{
+				// We want the connection to persist beyond the lifecycle of a single request
```

Comment:
Not sure I understand the significance of the comment. Does it relate to the `Context` line below?

Reply:
yep, not propagating the context from the request. I can clarify.